File: cos/cos.ops/src/main/java/cos/ops/StopMove.java
package cos.ops;

import java.nio.ByteBuffer;

public record StopMove(
        @Override byte code,
        @Override int id,
        @Override int userId,
        int x,
        int y,
        Direction sight
) implements AnyOp {

    public StopMove(int id, int userId, int x, int y, Direction sight) {
        this(Op.STOP_MOVE, id, userId, x, y, sight);
    }

    public static StopMove read(ByteBuffer b) {
        return new StopMove(
                b.getInt(),
                b.getInt(),
                b.getInt(),
                b.getInt(),
                Direction.of(b.get()));
    }

    public void write(ByteBuffer buf) {
        buf.putInt(id);
        buf.putInt(userId);
        buf.putInt(x);
        buf.putInt(y);
        buf.put((byte) sight.ordinal());
    }
}

File: cos/cos.olympus/src/main/kotlin/cos/olympus/game/ops/StopMove.kt
package cos.olympus.game.ops

import cos.olympus.Pos
import cos.olympus.game.Direction
import cos.olympus.game.IdDir

class StopMove(
    override val id: Int,
    val userId: Int,
    val x: Pos,
    val y: Pos,
    val dir: IdDir,
    val sight: IdDir
) : AnyOp
